build.rs

```diff
@@ -1,32 +1,43 @@
 use std::process::Command;
 
 fn current_git_hash() -> Option<String> {
-    let commit = Command::new("git")
-        .arg("log")
-        .arg("-1")
-        .arg("--pretty=format:%h") // Abbreviated commit hash
-        .current_dir(env!("CARGO_MANIFEST_DIR"))
-        .output();
-
-    if let Ok(commit) = commit {
-        if let Ok(commit) = String::from_utf8(commit.stdout) {
-            return Some(commit);
+    if option_env!("GIT_COMMIT") == None {
+        let commit = Command::new("git")
+            .arg("log")
+            .arg("-1")
+            .arg("--pretty=format:%h") // Abbreviated commit hash
+            .current_dir(env!("CARGO_MANIFEST_DIR"))
+            .output();
+    
+        if let Ok(commit) = commit {
+            if let Ok(commit) = String::from_utf8(commit.stdout) {
+                return Some(commit);
+            }
         }
+
+    } else {
+        return option_env!("GIT_COMMIT").map(String::from);
```

Comment:
Not sure if there's a better way to convert a `Option<&'static str>` to `Option<String>`.

Reply:
That's a perfect approach.